chia/simulator/block_tools.py

```diff
@@ -757,7 +757,12 @@ def get_consecutive_blocks(
                         if not use_timestamp_residual:
                             self._block_time_residual = 0.0
 
-                        full_block, block_record, self._block_time_residual = get_full_block_and_block_record(
+                        (
+                            full_block,
+                            block_record,
+                            self._block_time_residual,
+                            new_timestamp,
+                        ) = get_full_block_and_block_record(
```

Comment:
every new block returns its timestamp. This way, we increment `last_timestamp` for *every* block. Previously we would set it to the most recent tx-block.